samples/nextjs/src/components/Layout.tsx

```diff
@@ -43,10 +50,25 @@ const Navigation = () => {
 };
 
 type LayoutProps = {
-  route: RouteData;
+  layoutData: LayoutServiceData;
 };
 
-const Layout = ({ route }: LayoutProps): JSX.Element => {
+const Layout = ({ layoutData }: LayoutProps): JSX.Element => {
+  const { updateSitecoreContext } = useSitecoreContext({ updatable: true });
+
+  // Update Sitecore Context if layoutData has changed (i.e. on client-side route change).
+  // Note the context object type here matches the initial value in [[...path]].tsx.
+  useEffect(() => {
+    updateSitecoreContext &&
+      updateSitecoreContext({
+        route: layoutData.sitecore.route,
+        itemId: layoutData.sitecore.route.itemId,
+        ...layoutData.sitecore.context,
+      } as StyleguideSitecoreContextValue);
```

Comment:
@ambrauer For me it seems like better to define:
```
const context: StyleguideSitecoreContextValue = {
        route: layoutData.sitecore.route,
        itemId: layoutData.sitecore.route.itemId,
        ...layoutData.sitecore.context,
}

updateSitecoreContext && updateSitecoreContext(context)
```

Because `as` operator add ability to provide value which is not equal to type `StyleguideSitecoreContextValue`

Reply:
@illiakovalenko I agree, clearer too. Changed.